src/OptionSettlement.sol

```diff
@@ -18,35 +19,40 @@ import "./TokenURIGenerator.sol";
  * a broad swath of traditional options.
  */
 
-// TODO(DRY code during testing)
-// TODO(Gas Optimize)
-
-// @notice This settlement protocol does not support rebase tokens, or fee on transfer tokens
-
+/// @notice This settlement protocol does not support rebase tokens, or fee on transfer tokens
 contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
-    // The protocol fee
+    /// @notice The protocol fee
     uint8 public immutable feeBps = 5;
 
-    // The address fees accrue to
+    /// @notice The address fees accrue to
     address public feeTo = 0x36273803306a3C22bc848f8Db761e974697ece0d;
 
-    // Fee balance for a given token
+    /// @notice Fee balance for a given token
     mapping(address => uint256) public feeBalance;
 
-    // The list of claims for an option
-    mapping(uint160 => uint256[]) internal unexercisedClaimsByOption;
-
-    // Accessor for Option contract details
+    /// @notice Accessor for Option contract details
     mapping(uint160 => Option) internal _option;
 
-    // Accessor for claim ticket details
+    /// @notice Accessor for claim ticket details
     mapping(uint256 => Claim) internal _claim;
 
+    /// @notice Accessor for buckets of claims grouped by day
+    /// @dev This is to enable O(constant) time options exercise. When options are written,
+    /// the Claim struct in this mapping is updated to reflect the cumulative amount written
+    /// on the day in question. write() will add unexercised options into the bucket
+    /// corresponding to the # of days after the option type's creation.
+    /// exercise() will randomly assign exercise to a bucket <= the current day.
+    mapping(uint160 => mapping(uint16 => ClaimBucket)) internal _claimBucketByOptionAndDay;
+
+    /// @notice Accessor for mapping a claim id to its ClaimIndex
+    mapping(uint256 => ClaimIndex[]) internal _claimIdToClaimIndexArray;
+
     uint256 public hashMask = 0xFFFFFFFFFFFFFFFFFFFF000000000000;
 
+    uint256 internal constant ONE_DAY_SECONDS = 86400;
```

Comment:
It's better to use the included Solidity time unit literal `1 days` (which equals 86,400), both for security good practice and I believe would be a gas optimization